kw/json/encode.py

```diff
@@ -98,11 +98,55 @@ def modify_kwargs(kwargs):
         kwargs["default"] = partial(default_encoder, date_as_unix_time=date_as_unix_time)
 
 
+def format_value(value, precision):
+    """Format provided value."""
+    if isinstance(value, float):
+        return round(value, precision)
+    if isinstance(value, (list, set)):
+        return traverse_iterable(value, precision)
+    if isinstance(value, ItemsView) or value.__class__.__name__ == "dictionary-itemiterator":
+        return traverse_dict(dict(value), precision)
+    if isinstance(value, dict):
+        return traverse_dict(value, precision)
+    if isinstance(value, tuple):
+        if getattr(value, "_fields", False):
+            # namedtuple should stay namedtuple - simplejson can format it differently
+            # depending on the `namedtuple_as_object` param
+            NewNamedtuple = namedtuple(type(value).__name__, value._asdict().keys())
+            return NewNamedtuple(**format_value(value._asdict(), precision))
+        # convert tuple to list; the tuple would be turned to list in simplejson/json anyways
+        return traverse_iterable(list(value), precision)
+    return value
+
+
+def traverse_iterable(iterable, precision):
+    """Traverse list or set and round floats."""
+    new_iterable = [format_value(value, precision) for value in iterable]
+    return new_iterable if isinstance(iterable, list) else set(new_iterable)
```

Comment:
Why `set` is needed?

Reply:
I guess it is not needed since we would turn the `set` to `list` anyways when `simplejson` would raise `TypeError`.